src/OrchardCore.Modules/OrchardCore.Users/ConfigureCookieAuthenticationOptions.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+using OrchardCore.Settings;
+using OrchardCore.Users.Controllers;
+
+namespace OrchardCore.Users
+{
+    public class ConfigureCookieAuthenticationOptions : IConfigureNamedOptions<CookieAuthenticationOptions>
+    {
+        private readonly IServiceScopeFactory _serviceScopeFactory;
+
+        public ConfigureCookieAuthenticationOptions(IServiceScopeFactory serviceScopeFactory)
+        {
+            _serviceScopeFactory = serviceScopeFactory;
+        }
+
+        public void Configure(CookieAuthenticationOptions options)
+        {
+            Configure(null, options);
+        }
+
+        public void Configure(string name, CookieAuthenticationOptions options)
+        {
+            using (var scope = _serviceScopeFactory.CreateScope())
+            {
```

Comment:
Not good to create and then use a scope like this in a shell / tenant container, we would need to use our helpers to create a `ShellScope` and then use it in the corect way.

Anyway, because `ISiteService` is a tenant singleton, here you can and you should just inject it in the ctor.